run.py

```diff
@@ -269,7 +269,6 @@ def _validate_profile_options(profile_options: str):
     parser.add_argument("-t", "--test", choices=["eval", "train"], default="eval", help="Which test to run.")
     parser.add_argument("--profile", action="store_true", help="Run the profiler around the function")
     parser.add_argument("--profile-options", type=_validate_profile_options, help=f"Select which profile options to enable. Valid options: {SUPPORT_PROFILE_LIST}.")
-    parser.add_argument("--amp", action="store_true", help="enable torch.autocast()")
```

Comment:
Is the `--amp` option is being used by the PT2 compiler team, @jansel ? If so, we need to keep it.

Reply:
Yes, amp is used.

Reply:
Okay, I will keep `--amp` here, and try to use it in `extra_args.py` instead.

Reply:
In this PR, `--precision amp` in CUDA is `fp16`, in CPU is `bf16`. It is not extendable to support other low precision data types. for example, supporting `bf16` in CUDA or `fp16` in CPU.

I simply tried to move `amp` to outside args, and use `--precision` and `amp` together to determine which autocast precision will be used, however, the semantic of `amp` in various torchbenchmark models are `fp16` autocast, will encounter unexpected errors in CUDA test. I suppose the semantic of `amp` need to be unified to fit `amp` in CPU and GPU together. Will mark this as a `TODO` in the roadmap.